### PR TITLE
sensor provisioning

### PR DESCRIPTION
**Pull request by m-mcgowan**
# Problem Context

The gateway sends a provisioning event like

```
_health.qo {
    "method" : "sensor-provision"
    "text": "sensor-id"
}
```

These events are now recognized and stored in the database, with the primary effect that the gateway and nodes are created if they were not already present. The provisioning events are also stored, which could later be used to view a historical timeline of sensor provisioning.

This is a precursor to being able to display the device when the QR code is scanned.

## Changes

* The conversion of notehub to sparrow events - both routed and API events - takes care of recognizing this event format. Strings are not presently part of sensor data, and so an additional property `provisioned: 1` is added to the event, which becomes the reading for that event.  

* There are many other uses for `_health.qo` events, and we are only presently interested in provisioning events. A new property on the reading schema `prereq` is used as an additional filter to be sure only the matching events are stored as a provisioning reading. (`method=sensor-provision`)

#### Testing routed events:

 - [x] stand up a local development project, configured with the notehub project UID and auth token and route configured to your locally running instance of the app using an ephemeral database.  Run `yarn run db:init` to update the schema in the database and create the project. Verify the app loads and displays that there are no gateways. (If there are gateways it's due to an event being received, that's ok too.)

 - [x] Using the WebREPL, connect to the gateway Notecard and enter
```
{"cmd":"hub.log","method":"sensor-provision","text":"20323746323650050018000d","sync":true}
``` 
(This is the same command the sparrow gateway sends to the Notecard when a node is paired.)

- [x] verify in notehub that the event was received
- [x] refresh the app page and verify that a new node is present.  It will not have a name set and no readings, but it is present now in the database.
- [x] inspect the postgres database and verify that a new `provisioning` reading was added.

#### Testing Bulk Import

Since the event shapes are slightly different between routed events and bulk import (via the API), we must also test that

- [x] restart the ephemeral database server
- [x] run the bulk import process
- [x] inspect the server logs to validate that there were no import errors
- [x] verify that on the homepage there is a node present
- [x] inspect the postgres database and verify that a new `provisioning` reading was added.

e2e tests:

- [ ] Actually provision and pair a new sensor rather than emulating via the WebREPL.  I will do this as part of a larger testing effort once the QR codes are handled by the app.

### CODE REVIEW

**Comment by CarltonHenderson** [comment]
I'll look at this when I get back from my haircut...

**Comment by m-mcgowan** [comment]
[web app handles sensor provisioning events.](https://trello.com/c/aFFzlYIA/30-web-app-handles-sensor-provisioning-events)

**Review by CarltonHenderson — APPROVED** [review]
Simple! Nicely done.

File: src/services/prisma-datastore/db-init.ts [hunk 1]
```diff
@@ -16,7 +16,6 @@ import {
 
 const prisma = new PrismaClient();
```
Comment (CarltonHenderson) [inline src/services/prisma-datastore/db-init.ts:17]: To trigger this db change in production, we should change our netlify.toml. Thusly add a section like this

```toml
[context.production]
  command = "sed -i \"s|process.env.DEPLOY_URL|'${DEPLOY_URL}'|g\" config.ts && yarn build && yarn db:update"
```

that is, I added the `&& yarn db:update` to the default build command.

**Review by CarltonHenderson — APPROVED** [review]
more comments

File: __tests__/src/services/notehub/SparrowEvents.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { normalizeSparrowEvent } from "../../../../src/services/notehub/SparrowEvents";
+
+describe("SparrowEvents", () => {
+  it("normalizes sensor provisioning events", () => {
+    const body = {
```
Comment (CarltonHenderson) [inline __tests__/src/services/notehub/SparrowEvents.test.ts:5]:
```suggestion
blah
```

**Pull request merged**